### ISSUE TITLE
Built file in master requires fs

### ISSUE THREAD

**Issue by trygveaa** [mapbox/mapbox-gl-draw#970]
**mapbox-gl-js version**: 1.8.1
**mapbox-gl-draw version**: master

### Steps to Trigger Behavior

 1. Clone this repo and run `yarn && yarn build-min`
 2. Try to import `dist/mapbox-gl-draw.js` in a project

### Expected Behavior

It should work, without depending on `fs`.

### Actual Behavior

I can't use it because it depends on `fs`. That is, I'm getting:

```
ERROR in ./node_modules/@mapbox/mapbox-gl-draw/dist/mapbox-gl-draw.js
Module not found: Error: Can't resolve 'fs' in '/home/trygve/dev/my-project/front/node_modules/@mapbox/mapbox-gl-draw/dist'
```

If I try to do the same after checking out `v1.1.2` it works fine.

**Comment by Ecafracs** [comment]
I'm also encountering this issue.
It looks like mapbox-gl-draw depends on @mapbox/geojsonhint, which depends on zaach/jsonlint-lines, which is referring to fs.
@mapbox/geojsonhint is apparently "on pause" as of a year ago.
zaach/jsonlint-line was abandoned 4 years ago, and it looks like mapbox/jsonlint is a fork of a fork of it.

Can somebody who understand what these packages do figure out how to replace these dependencies to not rely on dead code?

**Comment by brendanmc6** [comment]
A fix would be much appreciated! @trygveaa would that PR you've linked fix this issue as well? Or would we need to wait for a downstream fix here?

As others have said, this seems to fix it for me:
`import MapboxDraw from '@mapbox/mapbox-gl-draw/dist/mapbox-gl-draw.js';`

Here are existing issues I have found:
https://github.com/mapbox/mapbox-gl-draw/issues/850
https://github.com/mapbox/mapbox-gl-draw/issues/844
https://github.com/mapbox/mapbox-gl-draw/issues/626

**Comment by trygveaa** [comment]
> A fix would be much appreciated! @trygveaa would that PR you've linked fix this issue as well? Or would we need to wait for a downstream fix here?

Yes, #977 fixes this issue. However I've written some comments for discussion in it, because I don't think bundling all the dependencies in the single js file is a good idea. But there are no downsides of #977 over the current situation.

> As others have said, this seems to fix it for me:
> `import MapboxDraw from '@mapbox/mapbox-gl-draw/dist/mapbox-gl-draw.js';`

That works because the issue I'm describing in this PR doesn't exist in version `v1.1.2`, which is what is currently on npm. However for master, i.e. the unreleased version, this solution does not work.

> Here are existing issues I have found:
> #850
> #844
> #626

To be clear, even though these are related, the issue I'm describing here is not the same as those issues. Those issues is about the main entry point for the package requiring `fs` and `path`. That can be worked around by importing the `dist` file instead as you mentioned. However, this issue is about that even the `dist` file requires `fs` and `path`, so this workaround does not work anymore.

PR #976 fixes what those issues describe, so you don't have to specify the `dist` file (but #977 is also required for that to work).

**Comment by georgir** [comment]
Now v1.2.0 is the current on npm and this module has become unusable by default because of this issue. I don't get how could it get out of beta and published like that.

**Comment by dougturnkey** [comment]
Hi!  I am a mapbox customer trying to use this tool for a product and I was wondering if this issue will be fixed soon?

**Comment by rpadilla6** [comment]
@dougturnkey I wouldn't wait for this to be patched, could take a minute. If you can live with the 1.1.2 version I downgraded to that and got it running without the fs error.

**Comment by csimpi** [comment]
Nothing?

**Comment by iacapuca** [comment]
Hey, any hope?

**Comment by ilancohen** [comment]
Is there any reason we couldn't just replace the usage of `geojsonhint` with this:
https://www.npmjs.com/package/geojson-validation

After all, `geojsonhint` is used in **only** one place, here:

```
  api.add = function (geojson) {
    const errors = geojsonhint.hint(geojson, { precisionWarning: false }).filter(e => e.level !== 'message');
    if (errors.length) {
      throw new Error(errors[0].message);
    }
```

**Comment by ericscottmarquez** [comment]
Yeah... why would this get published with this error...

**Comment by ericscottmarquez** [comment]
I'm having the same issue with fs as OP stated

**Comment by ericscottmarquez** [comment]
In reference to https://github.com/mapbox/mapbox-gl-draw/issues/626#issue-220964607,

Doing: 
```javascript
  node: {
    fs: "empty"
  }
```
really shouldn't be a valid solution...

**Comment by daveisfera** [comment]
Any chance we could get this fixed and released?

**Comment by neemanjabu** [comment]
For **webpack 5** use

```js
//webpack.config.js

module.exports = {
    ...
    resolve: {
        fallback: {
            "fs": false
        },
    }
}
```

**Comment by nrgapple** [comment]
`1.1.2` is still giving me the `fs` error

**Comment by ifzm** [comment]
+1 Can anyone help solve historical problems?

**Comment by kknight** [comment]
In case someone is struggling with this; on webpack 5 only the following helped:

```
resolve: {
    fallback: {
      "fs": false,
      "path": false
    },
}
```

**Comment by arindam1993** [comment]
👋 This should now be fixed with v1.2.1
